ArcGISRuntimeSDKQt_CppSamples/DisplayInformation/CreateSymbolStylesFromWebStyles/CreateSymbolStylesFromWebStyles.cpp

```diff
@@ -64,31 +63,31 @@ CreateSymbolStylesFromWebStyles::CreateSymbolStylesFromWebStyles(QObject* parent
   // Set the scale at which feature symbols and text will appear at their default size
   m_map->setReferenceScale(100'000);
 
-  createSymbolStyles();
+  performSymbolSearch();
 }
 
-void CreateSymbolStylesFromWebStyles::createSymbolStyles()
+void CreateSymbolStylesFromWebStyles::performSymbolSearch()
 {
   QMap<QString,QStringList> categoriesMap = createCategoriesMap();
-  SymbolStyle* symbolStyle = new SymbolStyle("Esri2DPointSymbolsStyle", {}, this);
   SymbolStyleSearchParameters searchParams;
+  searchParams.setKeys(categoriesMap.keys());
   searchParams.setKeysStrictlyMatch(true);
 
-  connect(symbolStyle, &SymbolStyle::searchSymbolsCompleted, this, [this, symbolStyle, categoriesMap](const QUuid& /* taskId */, SymbolStyleSearchResultListModel* searchResults)
+  SymbolStyle* symbolStyle = new SymbolStyle("Esri2DPointSymbolsStyle", {}, this);
+  symbolStyle->searchSymbolsAsync(searchParams).then(this,
+  [this, symbolStyle, categoriesMap](SymbolStyleSearchResultListModel* searchResults)
```

Comment:
I dont like the double pass through of categoriesMap, but couldnt find a quick way to refactor other than pulling categories map into a class field which I didnt think made sense. Any ideas?